src/btree/bt_curnext.c

```diff
@@ -904,9 +904,20 @@ __wt_btcur_next(WT_CURSOR_BTREE *cbt, bool truncating)
         if (page != NULL &&
           (cbt->page_deleted_count > WT_BTREE_DELETE_THRESHOLD ||
             (newpage && cbt->page_deleted_count > 0))) {
+            /* Signal the test thread. */
+            CONNECTION_CONTROL_POINT_DEFINE_TRIGGER(
+              session, WT_CONN_CONTROL_POINT_ID_WT_13450_TEST);
+            printf("Arriving at control point\n");
+            /* Wait here for the checkpoint thread. */
+            CONNECTION_CONTROL_POINT_SET_MATCH_VALUE_AND_WAIT(
+              session, WT_CONN_CONTROL_POINT_ID_WT_13450_CKPT, CUR2BT(cbt)->id);
+            printf("Past control point\n");
+
             /* If checkpoint is happening on the btree, we can only evict clean content. */
             if (__wt_btree_syncing_by_other_session(session)) {
                 if (!__wt_page_is_modified(page)) {
+                    printf("I'm going to crash now!!!\n");
+                    WT_ASSERT(session, false);
```

Comment:
As we're going to merge this to the project branch I'd suggest adding a statistic or something similar that can be read instead of crashing the test. Then we can verify we reached the code path externally.

Reply:
Done.  Added the below:
EvictStat('eviction_force_delete_in_checkpoint', 'forced eviction - btree being checkpointed and pages selected because of too many deleted items count'),